modules/rubiconBidAdapter.js

```diff
@@ -251,7 +251,8 @@ export const spec = {
       const eids = utils.deepAccess(bidderRequest, 'bids.0.userIdAsEids');
       if (eids && eids.length) {
         // filter out unsupported id systems
-        utils.deepSetValue(data, 'user.ext.eids', eids.filter(eid => ['adserver.org', 'pubcid.org', 'liveintent.com', 'liveramp.com', 'sharedid.org'].indexOf(eid.source) !== -1));
+        utils.deepSetValue(data, 'user.ext.eids', eids.filter(eid => ['adserver.org', 'pubcid.org', 'liveintent.com', 'liveramp.com', 'sharedid.org'].indexOf(eid.source) !== -1 ||
+          (eid.uids[0] && eid.uids[0].ext && eid.uids[0].ext.stype === 'ppuid')));
```

Comment:
The exchange team doesn't want to receive pubProvided IDs on the eids yet. Please revert this particular change and instead add to the block of code just below the comment "set user.id value from config value".

In that block, if `config.getConfig('user.id')` doesn't return anything, then look for the first `eid.uids[*].ext.stype === 'ppuid'` and set that value in the OpenRTB as user.id.

I think it would be safer to accept eid.uids[].ext.stype === 'ppuid' on any index of the uids array, not just the 0th element. I think the way the code is written here, if the publisher set two elements, with "stype:dmp" and "stype:ppuid" second, we wouldn't pick up the ppuid.